docs/book/src/developer/providers/migrations/v1.10-to-v1.11.md

```diff
@@ -3,80 +3,265 @@
+- The `GetIPFamily` method (deprecated since CAPI v1.8) has been removed
+
+### MachineDeployment
+
+- See changes that applies to [all kind](#all-kind)
+- The `spec.progressDeadlineSeconds` field (deprecated since CAPI v1.9) has been removed
+- Replica counters are now consistent with replica counters from other resources
+  - `status.readyReplicas` has now a new semantic based on machine's `Ready` condition
+  - `status.availableReplicas` has now a new semantic based on machine's `Available` condition
+  - `status.upToDateReplicas` has now a new semantic (and name) based on machine's `UpToDate` condition
+  - Temporarily, old replica counters will still be available under the `status.deprecated.v1beta1` struct
+- Support for terminal errors has been dropped (see [dataSecretCreated](#cluster-api-contract-changes)).
+  - `status.failureReason` and `status.failureMessage` will continue to exist temporarily under `status.deprecated.v1beta1`.
+  - The const values for `Failed` phase has been deprecated in the enum type for `machineDeployment.status.phase` (controllers are not setting this value anymore)
+  
+### MachineSet
+
+- See changes that applies to [all kind](#all-kind)
+- Replica counters are now consistent with replica counters from other resources
+  - `status.readyReplicas` has now a new semantic based on machine's `Ready` condition
+  - `status.availableReplicas` has now a new semantic based on machine's `Available` condition
+  - `status.upToDateReplicas` has now a new semantic (and name) based on machine's `UpToDate` condition
+  - Temporarily, old replica counters will still be available under the `status.deprecated.v1beta1` struct
+- Support for terminal errors has been dropped (see [dataSecretCreated](#cluster-api-contract-changes)).
+  - `status.failureReason` and `status.failureMessage` will continue to exist temporarily under `status.deprecated.v1beta1`.
+
+### MachinePool
+
+- See changes that applies to [all kind](#all-kind)
+- Replica counters are now consistent with replica counters from other resources
+  - `status.readyReplicas` has now a new semantic based on machine's `Ready` condition
+  - `status.availableReplicas` has now a new semantic based on machine's `Available` condition
+  - `status.upToDateReplicas` has now a new semantic (and name) based on machine's `UpToDate` condition
+  - Temporarily, old replica counters will still be available under the `status.deprecated.v1beta1` struct
+- Support for terminal errors has been dropped (see [dataSecretCreated](#cluster-api-contract-changes)).
+  - `status.failureReason` and `status.failureMessage` will continue to exist temporarily under `status.deprecated.v1beta1`.
+  - The const values for `Failed` phase has been deprecated in the enum type for `machinePool.status.phase` (controllers are not setting this value anymore)
+
+### Machine
```

Comment:
Is it ok not to write about minReadySeconds?

Reply:
I think we didn't merge any changes related to minReadySeconds yet

Reply:
Yeah, ideally each PR should document it's own bits, this reduces the chance of missing something